keel-ec2-plugin/src/main/kotlin/com/netflix/spinnaker/keel/api/ec2/ClusterSpec.kt

```diff
@@ -38,23 +33,26 @@ fun ClusterSpec.resolve(resolvedImages: ResolvedImages): Set<ServerGroup> =
   }
     .toSet()
 
-private fun ClusterSpec.resolveLaunchConfiguration(region: String, appVersion: String, imageId: String): LaunchConfiguration =
-  LaunchConfiguration(
-    appVersion = appVersion,
-    imageId = imageId,
-    instanceType = checkNotNull(overrides[region]?.launchConfiguration?.instanceType
+private fun ClusterSpec.resolveLaunchConfiguration(region: SubnetAwareRegionSpec): LaunchConfiguration {
+  val image = checkNotNull(overrides[region.region]?.launchConfiguration?.image
+    ?: defaults.launchConfiguration?.image) { "No image resolved / specified for ${region.region}" }
+  return LaunchConfiguration(
+    appVersion = image.appVersion,
+    imageId = image.id,
+    instanceType = checkNotNull(overrides[region.region]?.launchConfiguration?.instanceType
       ?: defaults.launchConfiguration?.instanceType),
-    ebsOptimized = checkNotNull(overrides[region]?.launchConfiguration?.ebsOptimized
+    ebsOptimized = checkNotNull(overrides[region.region]?.launchConfiguration?.ebsOptimized
```

Comment:
what does `region.region` means?

Reply:
It's a little awkward that the `ClusterRegion` object contains various properties, one of which is the region name. It used to be just a String before this change but now has other properties such as AZs, vpc subnet…